setup-tektoncd/common.sh

```diff
@@ -0,0 +1,72 @@
+#
+# Environment Variables with Default Values
+#
+
+# namespace name for the container registry
+readonly export REGISTRY_NAMESPACE="${REGISTRY_NAMESPACE:-registry}"
+# the container registry uses the internal k8s service hosntame
+readonly export REGISTRY_HOSTNAME="${REGISTRY_HOSTNAME:-registry.registry.svc.cluster.local}"
+
+# namespace name for Tekton Pipeline controller
+readonly export TEKTON_NAMESPACE="${TEKTON_NAMESPACE:-tekton-pipelines}"
+
+# timeout employed during rollout status and deployments in general
+readonly export DEPLOYMENT_TIMEOUT="${DEPLOYMENT_TIMEOUT:-5m}"
+
+#
+# Helper Functions
+#
+
+# print error message and exit on error.
+function fail() {
+    echo "ERROR: ${*}" >&2
+    exit 1
+}
+
+# print out a strutured message.
+function phase() {
+    echo "---> Phase: ${*}..."
+}
+
+# uses kubectl to check the deployment status on namespace and name informed.
+function rollout_status() {
+    local namespace="${1}"
+    local deployment="${2}"
+
+    if ! kubectl --namespace="${namespace}" --timeout=${DEPLOYMENT_TIMEOUT} \
+        rollout status deployment "${deployment}"; then
+        fail "'${namespace}/${deployment}' is not deployed as expected!"
+    fi
+}
+
+# inspect the path after the informed executable name.
+function probe_bin_on_path() {
+    local name="${1}"
+
+    if ! type -a ${name} >/dev/null 2>&1; then
+        fail "Can't find '${name}' on 'PATH=${PATH}'"
+    fi
+}
+
+# get the artifact url for the specific version (release) or latest.
+function get_release_artifact_url() {
+    local _org_repo="${1}"
+    local _version="${2}"
+
+    local _url="https://api.github.com/repos/${_org_repo}/releases"
+    if [[ "${_version}" == "latest" ]]; then
+        echo $(
+            curl -s ${_url}/latest |
+                jq -r '.assets[].browser_download_url' |
+                egrep -i 'release.yaml' |
+                head -n 1
```

Comment:
There could be more than one artifact matching. Most components have a tag/notag option at least - in this context either should work. Triggers have an extra "interceptors" file. 
See the logic in https://github.com/tektoncd/plumbing/blob/main/tekton/resources/cd/install-tekton-release.yaml#L58 for instance.

Reply:
It's fine for now as this only covers pipeline, but since this was added to a `common.sh` I guess the intention is for this to be extended to more projects (i.e. install triggers, chains, results, dashboard)

Reply:
yeah, we definitely need to enhance this.